### PR TITLE
HIVE-27651: Upgrade hbase version

### PR DESCRIPTION
**Pull request by zhangbutao**
### What changes were proposed in this pull request?

Upgrade hbase to latest stable version 2.5.6

### Why are the changes needed?

HIVE-26124 has tried to upgrade hbase to 2.4.11, but it failed with hadoop3 compatibility problem.
From https://hbase.apache.org/downloads.html, Hbase2.5.6 seems to be compatible with hadoop3, so let's try to upgrade hbase to 2.5.6.

### Does this PR introduce _any_ user-facing change?

No

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -143,7 +143,7 @@
     <hadoop.version>3.3.1</hadoop.version>
     <hadoop.bin.path>${basedir}/${hive.path.to.root}/testutils/hadoop</hadoop.bin.path>
     <hamcrest.version>1.3</hamcrest.version>
-    <hbase.version>2.0.0-alpha4</hbase.version>
+    <hbase.version>2.5.5</hbase.version>
```
Comment (ayushtkn) [inline pom.xml:146]: may be you need 2.5.5-hadoop3 here 
https://mvnrepository.com/artifact/org.apache.hbase/hbase-client/2.5.5-hadoop3

can try that, if the test passes post that

Comment (zhangbutao) [reply pom.xml]: Thanks for suggestion. I have used the 2.5.5 and 2.5.5-hadoop3 version, but both of the two failed.
I think Hbase2.5.5 has some compatibility issues with Hadoop3.3.1, and i have filed a Hbase ticket https://issues.apache.org/jira/browse/HBASE-28061. And i also find another ticket about Hbase2.5.5&Hadoop3.3.1 https://issues.apache.org/jira/browse/HBASE-28053.  I hope that can get some valuable info from Hbase community.

Comment (ayushtkn) [reply pom.xml]: Any pointers, here I see you said on the ticket hbase built with 3.3.1 works fine

Comment (ayushtkn) [reply pom.xml]: I think 2.5.6 is released, we can give that a try? HBASE-28061 was fixed in 2.5.6

Comment (zhangbutao) [reply pom.xml]: Sure, Changed hbase version to 2.5.6. Thanks for reminding!

**Comment by zhangbutao** [comment]
Luckily, got a green pass. :)

File: hbase-handler/src/test/results/negative/cascade_dbdrop.q.out [hunk 2]
```diff
@@ -18,8 +18,7 @@ TBLPROPERTIES ("hbase.table.name" = "hbase_table_0", "external.table.purge" = "t
 POSTHOOK: type: CREATETABLE
 POSTHOOK: Output: database:hbasedb
 POSTHOOK: Output: hbaseDB@hbase_table_0
-Found 3 items
-drwxr-xr-x   - ### USER ### ### GROUP ###          0 ### HDFS DATE ### hdfs://### HDFS PATH ###
+Found 2 items
```
Comment (ayushtkn) [inline hbase-handler/src/test/results/negative/cascade_dbdrop.q.out:21]: what is this change? One entry got missing?

**Review by ayushtkn — APPROVED** [review]
minor doubt, rest LGTM

File: hbase-handler/src/test/results/negative/cascade_dbdrop.q.out (hunk 2 above)
Comment (zhangbutao) [reply hbase-handler/src/test/results/negative/cascade_dbdrop.q.out:21]: Here i did a quick local test using hbase2.2.4 and hbase2.5.6:
I create a hbase table using **Hbase shell** command cli:
```
hbase:001:0> create 'mytable', 'cf1', 'cf2'
Created table mytable
```


- I checked the hdfs data directory of the table in Hbase 2.2.4 and found it has three directory
```
Found 3 items
drwxr-xr-x   - root supergroup          0 2023-10-29 16:16 hdfs://localhost:8028/hbase224/data/default/mytable/.tabledesc
drwxr-xr-x   - root supergroup          0 2023-10-29 16:16 hdfs://localhost:8028/hbase224/data/default/mytable/.tmp
drwxr-xr-x   - root supergroup          0 2023-10-29 16:16 hdfs://localhost:8028/hbase224/data/default/mytable/556dcd44aa1534942170a780383ad2d3
```
- I checked the hdfs data directory of the table in Hbase 2.5.6 and found it only has two directory
```
 Found 2 items
drwxr-xr-x   - root supergroup          0 2023-10-29 16:44 hdfs://localhost:8028/hbase256/data/default/mytable/.tabledesc
drwxr-xr-x   - root supergroup          0 2023-10-29 16:44 hdfs://localhost:8028/hbase256/data/default/mytable/50cac0c44862cd5f9e8387677b0cc0ad
```

It is quite clear that new Hbase version(2.5.6) has changed its data directory structure, and this This change has nothing to do with Apache Hive.

Do you think my this test can prove that this change has nothing to do with Apache Hive?

Comment (ayushtkn) [reply hbase-handler/src/test/results/negative/cascade_dbdrop.q.out:21]: Looks fine, Will merge later today

**Pull request merged**